src/codegen/compile.ml

```diff
@@ -5644,20 +5644,20 @@ module MakeSerialization (Strm : Stream) = struct
         get_ref_size ^^ G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^ set_ref_size ^^
         set_inc ^^ inc_data_size get_inc
       in
-      
+
       (* the incremental GC leaves array slice information in tag,
          the slice information can be removed and the tag reset to array
          as the GC can resume marking from the array beginning *)
       let clear_array_slicing =
         let (set_temp, get_temp) = new_local env "temp" in
-        set_temp ^^ 
+        set_temp ^^
         get_temp ^^ compile_unboxed_const Tagged.(int_of_tag StableSeen) ^^
         G.i (Compare (Wasm.Values.I32 I32Op.Ne)) ^^
         get_temp ^^ compile_unboxed_const Tagged.(int_of_tag CoercionFailure) ^^
         G.i (Compare (Wasm.Values.I32 I32Op.Ne)) ^^
         G.i (Binary (Wasm.Values.I32 I32Op.And)) ^^
         get_temp ^^ compile_unboxed_const Tagged.(int_of_tag ArraySliceMinimum) ^^
-        G.i (Compare (Wasm.Values.I32 I32Op.GeS)) ^^
+        G.i (Compare (Wasm.Values.I32 I32Op.GeS)) ^^ (*crusso: why signed? *)
```

Comment:
```suggestion
        G.i (Compare (Wasm.Values.I32 I32Op.GeS)) ^^ (* crusso: why signed? *)
```

Reply:
Thanks, good catch: It should be unsigned comparison.